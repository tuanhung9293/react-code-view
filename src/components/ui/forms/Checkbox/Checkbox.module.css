@value colors: 'styles/colors.css';
@value secondary, field from colors;
@value easings: 'styles/easings.css';
@value ease-out-cubic from easings;

.wrapper{
  flex-shrink: 0;
}

.input{
  width: 0;
  height: 0;
  opacity: 0;
}

.display{
  position: relative;
  display: inline-block;
  width: 20px;
  height: 20px;
  border: 2px solid transparent;
  background-color: field;
  box-shadow: 0 2px rgba(0,0,0,0.1);
  border-radius: 4px;
}

.display::after{
  content: "";
  background-image: url('../../../../assets/check-dark.svg');
  width: 14px;
  height: 11px;
  position: absolute;
  top: 5px;
  left: 3px;
  opacity: 0;
}

.input:focus + .display {
  border-color: secondary;
}

.input:checked + .display::after{
  opacity: 1;
}