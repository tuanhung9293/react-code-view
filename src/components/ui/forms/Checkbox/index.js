import React from 'react'
import styles from './Checkbox.module.css'

function Checkbox({hasError, ...props}) {
  return (
    <span className={styles.wrapper}>
      <input {...props} className={styles.input} type="checkbox" />
      <span className={styles.display} />
    </span>
  )
}

export default Checkbox