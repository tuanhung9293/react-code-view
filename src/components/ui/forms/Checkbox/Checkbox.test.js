import React from 'react'
import {shallow} from 'enzyme'
import Checkbox from './'

it('renders Checkbox', () => {
  expect(shallow(
    <Checkbox />
  )).toMatchSnapshot()
})

it('Checkbox should pass props down', () => {
  expect(shallow(
    <Checkbox data="data1" data2="data2" />
  )).toMatchSnapshot()
})
