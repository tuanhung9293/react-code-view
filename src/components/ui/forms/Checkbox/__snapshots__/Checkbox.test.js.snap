// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should pass props down 1`] = `
<span
  className="wrapper"
>
  <input
    className="input"
    data="data1"
    data2="data2"
    type="checkbox"
  />
  <span
    className="display"
  />
</span>
`;

exports[`renders Checkbox 1`] = `
<span
  className="wrapper"
>
  <input
    className="input"
    type="checkbox"
  />
  <span
    className="display"
  />
</span>
`;
